release/build.sh

```diff
@@ -14,133 +14,196 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-# Directory of this script
-SCRIPT_DIR="$( cd "$(dirname "$0")" ; pwd -P )"
-
 # Directory of the UDFs
 UDF_DIR=udfs
 
-# Location for new datasets
-LOCATION=US
-
 # Set colors if terminal supports it
 ncolors=$(tput colors)
-if [[ $ncolors -gt 7 ]]; then
+if [[ "${ncolors}" -gt 7 ]]; then
   BOLD=$(tput bold)
   NORMAL=$(tput sgr0)
 fi
 
 #######################################
-# Retrieves the dataset of a file by
-# evaluating the mapping of the file's
-# parent directory to an associated
-# dataset.
+# Executes a query file. If the execution
+# fails, the script will exit with an error
+# code of 1.
 # Globals:
-#   None
+#   BOLD
+#   NORMAL
+#   LOCATION
 # Arguments:
 #   file
+#   dry_run
 # Returns:
-#   The dataset
+#   None
 #######################################
-function get_dataset() {
+function execute_query() {
   local file=$1
-  local dataset=""
-  case $file in
-    *"community"*)
-      dataset="fn";;
-    *"netezza"*)
-      dataset="nz";;
-    *"oracle"*)
-      dataset="or";;
-    *"redshift"*)
-      dataset="rs";;
-    *"snowflake"*)
-      dataset="sf";;      
-    *"teradata"*)
-      dataset="td";;
-    *"vertica"*)
-      dataset="ve";;
-  esac
-  echo "$dataset"
-}
+  local dry_run=$2
 
+  printf "%s%s%s\n" "${BOLD}" "${file}" "${NORMAL}"
+  if [[ ${dry_run} = true ]]; then
+    if ! bq query \
+    --headless --nouse_legacy_sql --dry_run "$(cat "${file}")" ; then
+      printf "Failed to dry run: %s" "${file}"
+      # exit 1 is not called here because some dry-runs may fail due to
+      # variable placeholders which a user must replace with their own values.
+      # These dry-runs require manual revision of results.
+    fi
+  else
+    if ! bq query \
+    --headless --nouse_legacy_sql "$(cat "${file}")" ; then
+      printf "Failed to create: %s" "${file}"
+      exit 1
+    fi
+  fi
+}
 
 #######################################
-# Creates a dataset if it doesn't exist
-# already.
+# Builds and hosts the Cloud Build image
+# used to test BigQuery UDFs.
 # Globals:
-#   LOCATION
+#   UDF_DIR
 # Arguments:
-#   dataset
+#   None
 # Returns:
 #   None
 #######################################
-function create_dataset_if_not_exists() {
-  local dataset=$1
-  # If we failed to show the dataset, it doesn't exist so create it.
-  if ! bq show --headless $dataset > /dev/null 2>&1; then
-    bq mk --headless -d --data_location=$LOCATION $dataset
-  fi
+function build_udf_testing_image() {
+  gcloud builds submit "${UDF_DIR}"/tests/ \
+    --config="${UDF_DIR}"/tests/cloudbuild.yaml
```

Comment:
I think it might be ok to have multiple cloudbuild.yaml files since the directory path signals what it's building (e.g. the cloudbuild.yaml in udfs directory builds the udfs). 

reference: https://github.com/GoogleCloudPlatform/cloud-builders